tests/e2etest/GenericTest/src/main/java/org/astarteplatform/devicesdk/tests/e2etest/utilities/GenericMockDevice.java

```diff
@@ -0,0 +1,195 @@
+    }
+
+    interfacesDir = Paths.get("build/resources/standard-interfaces").toAbsolutePath().toString();
+
+    pairingUrl = apiUrl + "/pairing";
+
+    interfaceServerData = "org.astarte-platform.java.e2etest.ServerDatastream";
+    interfaceDeviceData = "org.astarte-platform.java.e2etest.DeviceDatastream";
+    interfaceServerAggr = "org.astarte-platform.java.e2etest.ServerAggregate";
+    interfaceDeviceAggr = "org.astarte-platform.java.e2etest.DeviceAggregate";
+    interfaceServerProp = "org.astarte-platform.java.e2etest.ServerProperty";
+    interfaceDeviceProp = "org.astarte-platform.java.e2etest.DeviceProperty";
+
+    byte[] mockByte = new byte[] {104, 101, 108, 108, 111};
+
+    DateTime dateTime1 = new DateTime("2023-09-13T21:39:45.618-08:00");
+
+    mockDataDictionary = new HashMap<>();
+    mockDataDictionary.put("double_endpoint", 5.4);
+    mockDataDictionary.put("integer_endpoint", 42);
+    mockDataDictionary.put("boolean_endpoint", true);
+    mockDataDictionary.put("longinteger_endpoint", 45543543534L);
+    mockDataDictionary.put("string_endpoint", "hello");
+    mockDataDictionary.put("binaryblob_endpoint", mockByte);
+    mockDataDictionary.put("datetime_endpoint", dateTime);
+    mockDataDictionary.put("doublearray_endpoint", new double[] {22.2, 322.22, 12.3, 0.1});
+    mockDataDictionary.put("integerarray_endpoint", new int[] {22, 322, 0, 10});
+    mockDataDictionary.put("booleanarray_endpoint", new boolean[] {true, false, true, false});
+    mockDataDictionary.put(
+        "longintegerarray_endpoint",
+        new long[] {45543543534L, 45543543534L, 45543543534L, 45543543534L});
+    mockDataDictionary.put("stringarray_endpoint", new String[] {"hello", " world"});
+    mockDataDictionary.put("binaryblobarray_endpoint", new byte[][] {mockByte, mockByte});
+    mockDataDictionary.put("datetimearray_endpoint", new DateTime[] {dateTime, dateTime1});
+  }
+
+  public String getRealm() {
+    return realm;
+  }
+
+  public String getDeviceId() {
+    return deviceId;
+  }
+
+  public String getCredentialsSecret() {
+    return credentialsSecret;
+  }
+
+  public String getApiUrl() {
+    return apiUrl;
+  }
+
+  public String getAppEngineToken() {
+    return appEngineToken;
+  }
+
+  public String getPairingUrl() {
+    return pairingUrl;
+  }
+
+  public String getInterfaceServerData() {
+    return interfaceServerData;
+  }
+
+  public String getInterfaceDeviceData() {
+    return interfaceDeviceData;
+  }
+
+  public String getInterfaceServerAggr() {
+    return interfaceServerAggr;
+  }
+
+  public String getInterfaceDeviceAggr() {
+    return interfaceDeviceAggr;
+  }
+
+  public String getInterfaceServerProp() {
+    return interfaceServerProp;
+  }
+
+  public String getInterfaceDeviceProp() {
+    return interfaceDeviceProp;
+  }
+
+  public Map<String, Object> getMockDataDictionary() {
+    return mockDataDictionary;
+  }
+
+  public boolean compereHasMas(Map<String, Object> deviceProperty) {
```

Comment:
Change the name of the function, to something that better represents what it does.